azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/tools/appservice/service/webapp/WebAppManagerImpl.java

```diff
@@ -0,0 +1,185 @@
+import com.azure.resourcemanager.appservice.models.DeploymentSlot;
+import com.azure.resourcemanager.appservice.models.JavaVersion;
+import com.azure.resourcemanager.appservice.models.RuntimeStack;
+import com.azure.resourcemanager.appservice.models.WebApp.Update;
+import com.azure.resourcemanager.appservice.models.WebContainer;
+import com.microsoft.azure.tools.appservice.AppService;
+import com.microsoft.azure.tools.appservice.model.DeployType;
+import com.microsoft.azure.tools.appservice.model.PublishingProfile;
+import com.microsoft.azure.tools.appservice.model.Runtime;
+import com.microsoft.azure.tools.appservice.model.WebApp;
+import com.microsoft.azure.tools.appservice.model.WebAppDeploymentSlot;
+import com.microsoft.azure.tools.appservice.service.AbstractAppServiceUpdatable;
+import com.microsoft.azure.tools.appservice.service.WebAppDeploymentSlotCreatable;
+import com.microsoft.azure.tools.appservice.service.WebAppManager;
+import com.microsoft.azure.tools.appservice.service.deploymentslot.WebAppDeploymentSlotCreatableImpl;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class WebAppManagerImpl implements WebAppManager {
+
+    private WebApp webApp;
+    private AppService appService;
+    private AzureResourceManager azureResourceManager;
+    private com.azure.resourcemanager.appservice.models.WebApp webAppService;
+
+    public WebAppManagerImpl(WebApp webApp, AppService appService) {
+        this.webApp = webApp;
+        this.appService = appService;
+        this.azureResourceManager = appService.getAzureResourceManager();
+    }
+
+    @Override
+    public void start() {
+        getWebAppService().start();
+    }
+
+    @Override
+    public void stop() {
+        getWebAppService().stop();
+    }
+
+    @Override
+    public void restart() {
+        getWebAppService().restart();
+    }
+
+    @Override
+    public void delete() {
+        azureResourceManager.webApps().deleteById(getWebAppService().id());
+    }
+
+    public void deploy(DeployType deployType, File target) {
+        getWebAppService().deploy(com.azure.resourcemanager.appservice.models.DeployType.fromString(deployType.getValue()), target);
+    }
+
+    @Override
+    public boolean exists() {
+        return getWebAppService(true) != null;
+    }
+
+    @Override
+    public PublishingProfile getPublishingProfile() {
+        final com.azure.resourcemanager.appservice.models.PublishingProfile publishingProfile = getWebAppService().getPublishingProfile();
+        return PublishingProfile.createFromServiceModel(publishingProfile);
+    }
+
+    @Override
+    public Runtime getRuntime() {
+        return Runtime.createFromServiceInstance(getWebAppService());
+    }
+
+    @Override
+    public WebAppUpdatable update(WebApp webApp) {
+        return new WebAppUpdatable();
+    }
+
+    @Override
+    public List<WebAppDeploymentSlot> listDeploymentSlots() {
```

Comment:
Suggest to move deploy and deploy slot to AeploymentManager: deployment().listDeploymentSlots